packages/x-components/src/plugins/__tests__/x-plugin-mixin.spec.ts

```diff
@@ -95,106 +77,149 @@ describe('testing $x component API global mixin', () => {
     });
   });
 
-  it('allows emitting and subscribing to events via $x object', () => {
-    const listener = jest.fn();
+  describe('testing $x API', () => {
+    it('allows emitting and subscribing to events via $x object', () => {
+      const [, localVue] = installNewXPlugin();
+      const wrapper = mount(
+        {
+          render(createElement) {
+            return createElement();
+          }
+        },
+        { localVue }
+      );
+      const listener = jest.fn();
 
-    componentInstance.vm.$x.on('UserIsTypingAQuery').subscribe(listener);
-    componentInstance.vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
+      wrapper.vm.$x.on('UserIsTypingAQuery').subscribe(listener);
+      wrapper.vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
 
-    expect(listener).toHaveBeenCalledTimes(1);
-    expect(listener).toHaveBeenCalledWith('So awesome, much quality, such skill');
-  });
+      expect(listener).toHaveBeenCalledTimes(1);
+      expect(listener).toHaveBeenCalledWith('So awesome, much quality, such skill');
+    });
+
+    it('emits the event metadata', () => {
+      const [, localVue] = installNewXPlugin();
+      const listener = jest.fn();
+      const testTarget = document.createElement('div');
 
-  it('emits the event metadata', () => {
-    const listener = jest.fn();
-    const testTarget = document.createElement('div');
-
-    const emitterComponent: ComponentOptions<any> & ThisType<Vue> = {
-      mounted() {
-        this.$x.emit('UserIsTypingAQuery', 'Sexy Lego', { target: testTarget });
-      },
-      render(createElement) {
-        return createElement('input');
-      }
-    };
-
-    mount(
-      {
-        mixins: [xComponentMixin(searchBoxXModule)],
-        created() {
-          this.$x.on('UserIsTypingAQuery', true).subscribe(listener);
+      const emitterComponent: ComponentOptions<any> & ThisType<Vue> = {
+        mounted() {
+          this.$x.emit('UserIsTypingAQuery', 'Sexy Lego', { target: testTarget });
```

Comment:
Who is the mentally ill developer that decided `Sexy Lego` was a proper query to test with? 🤣

Reply:
jijiijij

Reply:
I see you complain about Sexy Lego, but you said nothing about Sexy playmobil, are you a playmobil fanboy?